frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx

```diff
@@ -71,20 +67,43 @@ class DashboardInner extends Component {
       return;
     }
 
+    const isParameterValuesChanged = !_.isEqual(
+      prevProps.parameterValues,
+      this.props.parameterValues,
+    );
+    // This is a way of detecting card parameter mapping changes. Parameter
+    // mappings change happen independently of isEditing because of re-renders,
+    // so we use this heuristic to avoid additional state. The fetching logic
+    // should be moved outside.
+    const isDirtyChanged = !_.isEqual(
+      this._getDashcardsIsDirty(prevProps.dashboard?.dashcards),
+      this._getDashcardsIsDirty(this.props.dashboard?.dashcards),
+    );
```

Comment:
At first glance, this doesn't look so different from `!_.isEqual(prevDashcard, dashcard)`, does it?
We wanted to avoid that kind of comparison, so minor changes to a dashcard (like something purely visual) won't trigger query reruns. IIRC, `isDirty: true` should be set on a dashcard after _every_ edit, so this still feels like a more optional `_.isEqual(prevDashcard, dashcard)`. Does that make sense or did I miss something?

Reply:
@oleggromov should we compare only `parameter_mappings`? Something like this:
```
getParameterMappings(dashboard) {
    return dashboard.dashcards.map(dashcard => dashcard.parameter_mappings);
}

....
if (!_.isEqual(
  getParameterMappings(this.props.dashboard)),
  getParameterMappings(prevProps.dashboard))
) {
  this.props.fetchDashboardCardData({ reload: false, clearCache: true });
}
```

Reply:
https://github.com/metabase/metabase/pull/36657#issuecomment-1852265577

Reply:
@kulyk wdyt? Based on the original issue I think it's what we want to handle

Reply:
We did explore it with @ranquild and I hope we've found a better fix within the `UPDATE_DASHBOARD_AND_CARDS` action. I'll make a new version of the fix and submit it for review soon.
Thank you.